### PR TITLE
chore: Test more authentication methods

### PR DESCRIPTION
**Pull request by sfc-gh-jmichalak**
- add a test for JWT authentication (skipped for now; it needs a configuration adjustment and this will be done in [the next PR](https://github.com/Snowflake-Labs/terraform-provider-snowflake/pull/3180))
- add a test for Okta
- add a test for session parameters
- move driver log level to sdk
- add tests for Mfa with and without passcode
- fix conflicting fields in view resource
- bring back configuring logging level for internal client tests
- describe all of the manual test steps in a readme

## Test Plan

* [x] acceptance tests

* [ ] …

## TODO

- [(next pr)](https://github.com/Snowflake-Labs/terraform-provider-snowflake/pull/3180) read a compatible config file for older versions in acc tests, adjust the configurations in Github secrets and our secret storage, and unskip the relevant tests
- Other authentication methods will be done in SNOW-1791729.

### CODE REVIEW

File: pkg/provider/provider_acceptance_test.go [hunk 1]
```diff
@@ -543,6 +543,84 @@ func TestAcc_Provider_triValueBoolean(t *testing.T) {
...
+		},
+		Steps: []resource.TestStep{
+			// authenticate with unencrypted private key
+			{
+				Config: providerConfigWithAuthenticator("jwt_test", sdk.AuthenticationTypeJwt),
+			},
+			// authenticate with unencrypted private key with a legacy authenticator value
+			// solves https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/2983
+			{
+				Config: providerConfigWithAuthenticator("jwt_test", sdk.AuthenticationTypeJwtLegacy),
```
Comment (sfc-gh-jmichalak) [inline pkg/provider/provider_acceptance_test.go:595]: This is moved to `testprofiles` in the next pr.

File: pkg/provider/manual_tests/README.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
+# Manual tests
+
+This directory is dedicated to hold steps for manual provider tests that are not possible to re-recreate in automated tests (or very hard to set up). These tests are disabled by default and require `TEST_SF_TF_ENABLE_MANUAL_TESTS` environmental variable to be set.
+
+## Okta authenticator test
+This test checks `Okta` authenticator option. It requires manual steps because of additional setup on Okta side. It assumes that `default` profile uses a standard values of account name, user, password, etc.
+1. Set up a developer Okta account [here](https://developer.okta.com/signup/).
+1. Go to admin panel and select Applications -> Create App Integration.
```
[the remark this answers is not in the document]
Comment (sfc-gh-jcieslak) [reply pkg/provider/manual_tests/README.md:8]: It seems it's rendered correctly, but after SQL block the counting started again, so you may want to exactly specify the ordering.

[the remark this answers is not in the document]
Comment (sfc-gh-jmichalak) [reply pkg/provider/manual_tests/README.md]: Well, it's a Markdown feature so you don't have to change the number every time you move/add/delete lines, but it breaks on code blocks, I guess. Fixed.

File: pkg/sdk/internal/client/client_test.go [hunk 3]
```diff
@@ -21,6 +21,7 @@ func TestNewClientWithoutInstrumentedSQL(t *testing.T) {
 
 	t.Run("registers snowflake-not-instrumented driver", func(t *testing.T) {
 		config := sdk.DefaultConfig()
+		config.Tracing = string(sdk.DriverLogLevelDebug)
```
[the remark this answers is not in the document]
Comment (sfc-gh-jmichalak) [reply pkg/sdk/internal/client/client_test.go:24]: Added to acc tests.

File: pkg/sdk/client.go [hunk 4]
```diff
@@ -21,6 +21,7 @@ var (
 
 func init() {
 	instrumentedSQL = os.Getenv(snowflakeenvs.NoInstrumentedSql) == ""
+	gosnowflakeLoggingLevel = os.Getenv(snowflakeenvs.DriverTracing)
```
[the remark this answers is not in the document]
Comment (sfc-gh-jmichalak) [reply pkg/sdk/client.go]: Removed.

File: pkg/provider/manual_tests/README.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
+# Manual tests
```
[the remark this answers is not in the document]
Comment (sfc-gh-jmichalak) [reply pkg/provider/manual_tests/README.md]: Done.

**Review by sfc-gh-jcieslak — APPROVED** [review]

**Pull request merged**